baseplate/secrets/store.py

```diff
@@ -0,0 +1,253 @@
+"""Application integration with the secret fetcher daemon."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import base64
+import binascii
+import collections
+import datetime
+import json
+import logging
+
+from .. import config
+from ..context import ContextFactory
+
+
+ISO_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+logger = logging.getLogger(__name__)
+
+
+class SecretNotFoundError(Exception):
+    """Raised when the requested secret is not in the local vault."""
+
+    def __init__(self, name):
+        super(SecretNotFoundError, self).__init__()
+        self.name = name
+
+    def __str__(self):  # pragma: nocover
+        return "secret not found: {!r}".format(self.name)
+
+
+class CorruptSecretError(Exception):
+    """Raised when the requested secret does not match the expected format."""
+
+    def __init__(self, path, message):
+        super(CorruptSecretError, self).__init__()
+        self.path = path
+        self.message = message
+
+    def __str__(self):  # pragma: nocover
+        return "{}: {}".format(self.path, self.message)
+
+
+_VersionedSecret = collections.namedtuple(
+    "VersionedSecret", "previous current next")
+
+
+class VersionedSecret(_VersionedSecret):
```

Comment:
Negative. This is purely our convention. I'd like to get a cron job that rotates keys going but that'd be vault-side so apps don't even need to know it exists as long as they can handle versioned secrets OK.